### ISSUE TITLE
Grid not working as expected

### ISSUE THREAD

**Issue by oliverdolgener** [mobily/stacks#28]
First of all thank you so much for this awesome library.
We're starting to use it in our production app and we're already loving it.

We notice some unexpected behaviour while playing around with the `Grid` component.

- Unfortunately it does not use the same multiplied values as all the other components so you have to pass in the 'real' values.

- Another thing is that the `margin` prop only adds to the right edge. The left edge is using the `gutter` value.

- It would also be very useful to provide multiple values for all three breakpoints (e.g 4 columns on mobile, 8 on tablet, 12 on desktop as well as margins and gutters)

That makes it a bit hard to work with. Maybe you could have a look.

We use it like that:
```
<StacksProvider spacing={4} breakpoints={{ tablet: 600, desktop: 940 }}>
  {... Navigators & Stuff}
  <Grid columns={4} gutter={8} margin={16} />
</StacksProvider>
```

<img src="https://user-images.githubusercontent.com/22632842/123078693-81b10680-d41b-11eb-8b0b-12c9b16406b4.png" height="600">

As you can see it applies 16px of `margin` to the right side but uses 8px of `gutter` for the left side.
We were expecting `margin` to be applied to both sides.

Thanks in advance!

**Comment by mobily** [comment]
@oliverdolgener hey! thanks for reporting this issue :) to be honest, after rewriting to ReScript I didn't check if the Grid component works fine 🙈 anyway, after the initial tests, please be aware that for some screen resolutions, some Grid configs don't work as intended (I think I should add more detailed info in the docs)

**Comment by mobily** [comment]
@oliverdolgener 👋 Grid calculations have been fixed in v1.3.1, now it accepts the `ResponsiveProp` format and multiplied values (instead of "real" ones) 😊 see the screenshot below, I'm using the same values as you provided in the original post: `<Grid columns={4} gutter={8} margin={16} />`

<img width="445" alt="CleanShot 2021-09-19 at 12 06 01@2x" src="https://user-images.githubusercontent.com/1467712/133923787-3cf6e140-df0a-4348-8366-9ccb5743c6e2.png">
